src/Controller/EnMarche/CitizenActionController.php

```diff
@@ -2,31 +2,106 @@
 
 namespace AppBundle\Controller\EnMarche;
 
+use AppBundle\CitizenAction\CitizenActionManager;
+use AppBundle\CitizenAction\CitizenActionRegistrationCommandHandler;
+use AppBundle\Controller\CanaryControllerTrait;
 use AppBundle\Controller\EntityControllerTrait;
 use AppBundle\Entity\CitizenAction;
+use AppBundle\Event\EventRegistrationCommand;
+use AppBundle\Exception\BadUuidRequestException;
+use AppBundle\Exception\InvalidUuidException;
+use AppBundle\Form\EventRegistrationType;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Entity;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpFoundation\ResponseHeaderBag;
 
 /**
  * @Route("/action-citoyenne")
- * @Entity("action", expr="repository.findOnePublishedBySlug(slug)")
+ * @Entity("action", expr="repository.findOneCitizenActionBySlug(slug)")
  */
 class CitizenActionController extends Controller
 {
     use EntityControllerTrait;
+    use CanaryControllerTrait;
 
     /**
      * @Route("/{slug}", name="app_citizen_action_show")
      * @Method("GET")
      */
-    public function showAction(CitizenAction $action): Response
+    public function showAction(CitizenAction $action, CitizenActionManager $citizenActionManager): Response
     {
         return $this->render('citizen_action/show.html.twig', [
             'citizen_action' => $action,
+            'participants' => $citizenActionManager->getRegistrations($action),
+        ]);
+    }
+
+    /**
+     * @Route("/{slug}/inscription", name="app_citizen_action_attend")
+     * @Method("GET|POST")
+     */
+    public function attendAction(Request $request, CitizenAction $citizenAction): Response
+    {
+        if ($citizenAction->isFinished()) {
+            throw $this->createNotFoundException(sprintf('Event "%s" is finished and does not accept registrations anymore', $citizenAction->getUuid()));
+        }
+
+        if ($citizenAction->isCancelled()) {
+            throw $this->createNotFoundException(sprintf('Event "%s" is cancelled and does not accept registrations anymore', $citizenAction->getUuid()));
+        }
+
+        $command = new EventRegistrationCommand($citizenAction, $this->getUser());
+        $form = $this->createForm(EventRegistrationType::class, $command);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->get(CitizenActionRegistrationCommandHandler::class)->handle($command);
+            $this->addFlash('info', $this->get('translator')->trans('citizen_initiative.registration.success'));
```

Comment:
i guess it shouldn't be a citizen_`initiative` anymore

Reply:
`citizen_initiative.registration.success: Votre inscription est confirmée.` is universel